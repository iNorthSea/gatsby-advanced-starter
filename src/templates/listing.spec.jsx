import React from "react";
import { render } from "@testing-library/react";
import ListingPage from "./listing";
import ConfigContext from "../context/ConfigContext";
import config from "../../data/SiteConfig";

const { allMarkdownRemark } = require("../../test/sampleData");

const renderPaging = (postsPerPage = 3, currentPage = 1) => {
  const pageCount = Math.ceil(allMarkdownRemark.edges.length / postsPerPage);

  const pageContext = {
    limit: postsPerPage,
    skip: (currentPage - 1) * postsPerPage,
    pageCount,
    currentPageNum: currentPage,
  };
  const data = {
    allMarkdownRemark: {
      edges: allMarkdownRemark.edges.slice(
        pageContext.skip,
        pageContext.skip + pageContext.limit
      ),
    },
  };

  return render(
    <ConfigContext.Provider value={{ ...config, postsPerPage }}>
      <ListingPage pageContext={pageContext} data={data} />
    </ConfigContext.Provider>
  );
};

describe("page component ListingPage", () => {
  it("renders correctly with paging", () => {
    expect.assertions(3);

    const firstPage = renderPaging(3, 1);
    expect(firstPage.asFragment()).toMatchSnapshot();

    const secondPage = renderPaging(3, 2);
    expect(secondPage.asFragment()).toMatchSnapshot();

    const lastPage = renderPaging(3, 3);
    expect(lastPage.asFragment()).toMatchSnapshot();
  });

  it("renders correctly without paging", () => {
    expect.assertions(1);

    const pageContext = {
      limit: allMarkdownRemark.edges.length,
      skip: 0,
      pageCount: 1,
      currentPageNum: 1,
    };
    const data = {
      allMarkdownRemark,
    };

    const { asFragment } = render(
      <ConfigContext.Provider value={{ ...config, postsPerPage: 0 }}>
        <ListingPage pageContext={pageContext} data={data} />
      </ConfigContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
