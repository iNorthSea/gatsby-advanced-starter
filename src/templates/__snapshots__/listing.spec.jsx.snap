// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page component ListingPage renders correctly with paging 1`] = `
<DocumentFragment>
  <div
    class="layout-container"
  >
    <div
      class="listing-container"
    >
      <div
        class="posts-container"
      >
        <div>
          <a
            href="/big-sample-test"
          >
            <h1>
              Big Test
            </h1>
          </a>
          <a
            href="/the-fairys-witches"
          >
            <h1>
              The Fairy's Witches
            </h1>
          </a>
          <a
            href="/the-butterfly-of-the-edge"
          >
            <h1>
              The Butterfly of the Edge
            </h1>
          </a>
        </div>
      </div>
      <div
        class="paging-container"
      >
        <a
          href="/"
        >
          1
        </a>
        <a
          href="/2/"
        >
          2
        </a>
        <a
          href="/3/"
        >
          3
        </a>
        <a
          href="/2/"
        >
          Next
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`page component ListingPage renders correctly with paging 2`] = `
<DocumentFragment>
  <div
    class="layout-container"
  >
    <div
      class="listing-container"
    >
      <div
        class="posts-container"
      >
        <div>
          <a
            href="/random-project-4-with-super-long-title-phase-one"
          >
            <h1>
              Random project 4 With Super Long Title Phase One
            </h1>
          </a>
          <a
            href="/angels-of-mist"
          >
            <h1>
              Angels of Mist
            </h1>
          </a>
          <a
            href="/birch-in-the-roses"
          >
            <h1>
              Birch in the Roses
            </h1>
          </a>
        </div>
      </div>
      <div
        class="paging-container"
      >
        <a
          href="/"
        >
          Previous
        </a>
        <a
          href="/"
        >
          1
        </a>
        <a
          href="/2/"
        >
          2
        </a>
        <a
          href="/3/"
        >
          3
        </a>
        <a
          href="/3/"
        >
          Next
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`page component ListingPage renders correctly with paging 3`] = `
<DocumentFragment>
  <div
    class="layout-container"
  >
    <div
      class="listing-container"
    >
      <div
        class="posts-container"
      >
        <div>
          <a
            href="/the-fallen-time"
          >
            <h1>
              The Fallen Time
            </h1>
          </a>
          <a
            href="/bold-mage"
          >
            <h1>
              Bold Mage
            </h1>
          </a>
        </div>
      </div>
      <div
        class="paging-container"
      >
        <a
          href="/2/"
        >
          Previous
        </a>
        <a
          href="/"
        >
          1
        </a>
        <a
          href="/2/"
        >
          2
        </a>
        <a
          href="/3/"
        >
          3
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`page component ListingPage renders correctly without paging 1`] = `
<DocumentFragment>
  <div
    class="layout-container"
  >
    <div
      class="listing-container"
    >
      <div
        class="posts-container"
      >
        <div>
          <a
            href="/big-sample-test"
          >
            <h1>
              Big Test
            </h1>
          </a>
          <a
            href="/the-fairys-witches"
          >
            <h1>
              The Fairy's Witches
            </h1>
          </a>
          <a
            href="/the-butterfly-of-the-edge"
          >
            <h1>
              The Butterfly of the Edge
            </h1>
          </a>
          <a
            href="/random-project-4-with-super-long-title-phase-one"
          >
            <h1>
              Random project 4 With Super Long Title Phase One
            </h1>
          </a>
          <a
            href="/angels-of-mist"
          >
            <h1>
              Angels of Mist
            </h1>
          </a>
          <a
            href="/birch-in-the-roses"
          >
            <h1>
              Birch in the Roses
            </h1>
          </a>
          <a
            href="/the-fallen-time"
          >
            <h1>
              The Fallen Time
            </h1>
          </a>
          <a
            href="/bold-mage"
          >
            <h1>
              Bold Mage
            </h1>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
